### PR TITLE
Replace Balance type with Decimal

### PR DESCRIPTION
**Pull request by Gauthamastro**
Changes the Balance handling inside ocex to Decimal instead of Balance

### CODE REVIEW

File: pallets/ocex/src/lib.rs [hunk 1]
```diff
@@ -360,12 +359,19 @@ pub mod pallet {
...
 				base_asset: base,
 				quote_asset: quote,
-				min_price: min_order_price,
-				max_price: max_order_price,
-				price_tick_size,
-				min_qty: min_order_qty,
-				max_qty: max_order_qty,
-				qty_step_size,
+				min_price: Decimal::from(min_order_price.saturated_into::<u128>())
+					.div(&Decimal::from(UNIT_BALANCE)),
```
Comment (serhii-temchenko) [inline pallets/ocex/src/lib.rs:368]: Is it okay to use here potentially `panic`ing code? `.div()` can panic ([here](https://github.com/paupino/rust-decimal/blob/master/src/arithmetic_impls.rs#L218-L219)). I see that here used constant value (e.g. `UNIT_BALANCE`) but still...

File: pallets/ocex/src/tests.rs [hunk 2]
```diff
@@ -664,25 +657,20 @@ fn test_submit_snapshot_sender_is_not_attested_enclave() {
...
+				withdrawals: Default::default(),
+				fees: bounded_vec![],
+			};
 		assert_noop!(
 			OCEX::submit_snapshot(Origin::signed(account_id.into()), snapshot, sig.clone().into()),
 			Error::<Test>::SenderIsNotAttestedEnclave
 		);
 		// There is an existing ingress message which holds RegisterUser
-		assert_eq!(OCEX::ingress_messages().len(), 1);
+		assert_eq!(OCEX::ingress_messages().len(), 0);
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs]: Please remove/update comment

File: pallets/polkadex-ido/src/lib.rs [hunk 3]
```diff
@@ -101,11 +101,11 @@ pub mod weights;
...
-#[cfg(test)]
-mod mock;
-
-#[cfg(test)]
-mod test;
+// #[cfg(test)]
+// mod mock;
+//
+// #[cfg(test)]
+// mod test;
```
Comment (serhii-temchenko) [inline pallets/polkadex-ido/src/lib.rs]: Why is this commented out?

File: pallets/test-token-provider/src/lib.rs [hunk 4]
```diff
@@ -1,10 +1,10 @@
...
-#[cfg(test)]
-mod mock;
-
-#[cfg(test)]
-mod test;
+// #[cfg(test)]
+// mod mock;
+//
+// #[cfg(test)]
+// mod test;
```
Comment (serhii-temchenko) [inline pallets/test-token-provider/src/lib.rs]: Same

File: pallets/ocex/src/tests.rs [hunk 5]
```diff
@@ -1240,7 +1142,7 @@ pub fn create_withdrawal_500<T: Config>(
 	return withdrawal
 }
 
-pub fn create_fees<T: Config>() -> Fees<BalanceOf<T>> {
-	let fees: Fees<BalanceOf<T>> = Fees { asset: AssetId::polkadex, amount: 100_u32.into() };
+pub fn create_fees<T: Config>() -> Fees {
+	let fees: Fees = Fees { asset: AssetId::polkadex, amount: Decimal::new(100, 1) };
 	return fees
 }
```
Comment (serhii-temchenko) [inline pallets/ocex/src/tests.rs:1212]: No tests to cover `FailedToConvertDecimaltoBalance` error.

**Review by serhii-temchenko — CHANGES_REQUESTED** [review]

File: pallets/ocex/src/lib.rs [hunk 6]
```diff
@@ -159,6 +157,7 @@ pub mod pallet {
 
 	#[pallet::error]
 	pub enum Error<T> {
+		FailedToConvertDecimaltoBalance,
```
Comment (35359595) [inline pallets/ocex/src/lib.rs:164]: please add document desctiption - it's converted into message on polkadot/UI side

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (35359595) [reply pallets/ocex/src/lib.rs:368]: agree, two checks are required: wount overflow + not divided by zero prior to each op

Comment (35359595) [reply pallets/ocex/src/lib.rs:368]: also interesting how 96 bit Decimal takes in 128 bit integer as saturated_into from very small and very big values, which potentially can underflow/overflow

**Review by 35359595 — CHANGES_REQUESTED** [review]

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (Gauthamastro) [reply pallets/ocex/src/lib.rs:368]: Resolved

File: pallets/polkadex-ido/src/lib.rs (hunk 3 above)
Comment (Gauthamastro) [reply pallets/polkadex-ido/src/lib.rs]: IDO and faucet will be removed from develop

File: pallets/ocex/src/lib.rs (hunk 1 above)
Comment (serhii-temchenko) [reply pallets/ocex/src/lib.rs:368]: I can see that this was resolved for the `deposit()` method in [this](https://github.com/Polkadex-Substrate/Polkadex/pull/518) PR, but not here.

Comment (serhii-temchenko) [reply pallets/ocex/src/lib.rs:368]: Had discussion with Ivan, he explained. Resolving.

**Pull request merged**